aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -173,6 +173,35 @@ void clamp_max_kernel_cuda(TensorIterator& iter, Scalar max_value) {
   });
 }
 
+void heaviside_tensor_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "heaviside_tensor_cuda", [&]() {
+    gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t self, scalar_t val) -> float {
+       if (self > 0) {
+         return 1.0;
+       } else if (self < 0) {
+         return 0.0;
+       } else {
+         return val;
+       }
+    });
+  });
+}
+
+void heaviside_scalar_kernel_cuda(TensorIterator& iter, Scalar val) {
```

Comment:
Probably don't need this kernel.

Reply:
Same, we no longer need  a separate kernel for scalar val, deleted.